packages/salesforcedx-apex-debugger/src/commands/requestService.ts

```diff
@@ -73,34 +75,40 @@ export class RequestService {
     this._proxyAuthorization = proxyAuthorization;
   }
 
+  public get connectionTimeoutMs(): number {
+    return this._connectionTimeoutMs || DEFAULT_CONNECTION_TIMEOUT_MS;
+  }
+
+  public set connectionTimeoutMs(connectionTimeoutMs: number) {
+    this._connectionTimeoutMs = connectionTimeoutMs;
+  }
+
   public async execute(command: BaseCommand): Promise<string> {
-    configure(this._proxyUrl, this._proxyStrictSSL);
+    if (this.proxyUrl) {
+      configure(this._proxyUrl, this._proxyStrictSSL);
+    }
     const urlElements = [this.instanceUrl, command.getCommandUrl()];
     const requestUrl =
       command.getQueryString() == null
         ? urlElements.join('/')
         : urlElements.join('/').concat('?', command.getQueryString()!);
-    const options: XHROptions =
-      command.getRequest() == null
-        ? {
-            type: 'POST',
-            url: requestUrl,
-            headers: {
-              'Content-Type': 'application/json',
-              Accept: 'application/json',
-              Authorization: `OAuth ${this.accessToken}`
-            }
-          }
-        : {
-            type: 'POST',
-            url: requestUrl,
-            headers: {
-              'Content-Type': 'application/json',
-              Accept: 'application/json',
-              Authorization: `OAuth ${this.accessToken}`
-            },
-            data: JSON.stringify(command.getRequest())
-          };
+    const requestBody = command.getRequest()
+      ? JSON.stringify(command.getRequest())
+      : undefined;
+    const options: XHROptions = {
+      type: 'POST',
+      url: requestUrl,
+      timeout: this.connectionTimeoutMs,
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+        Accept: 'application/json',
+        Authorization: `OAuth ${this.accessToken}`,
+        'Content-Length': requestBody
```

Comment:
@vazexqi This was the difference between sandbox and local. Fixes W-4360024.

Reply:
The missing content-length? Wow, some of the specs are really strict.